src/__tests__/cleanup.js

```diff
@@ -31,7 +31,7 @@ test('cleanup waits for queued microtasks during unmount sequence', async () =>
   const spy = jest.fn()
 
   const Test = () => {
-    React.useEffect(() => () => setImmediate(spy))
+    React.useEffect(() => () => setTimeout(spy, 200))
```

Comment:
Users can do all kinds of async things as a result of their cleanup functions and it's not reasonable for them to expect that we'll wait until all those async things are done. They'll have to wait for those things themselves.

So we shouldn't have a `setImmediate` in here at all.